src/main/java/org/venice/beachfront/bfapi/config/OAuth2Config.java

```diff
@@ -0,0 +1,199 @@
+import org.springframework.security.oauth2.client.token.grant.code.AuthorizationCodeAccessTokenProvider;
+import org.springframework.security.oauth2.client.token.grant.implicit.ImplicitAccessTokenProvider;
+import org.springframework.security.oauth2.client.token.grant.password.ResourceOwnerPasswordAccessTokenProvider;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.util.UriComponentsBuilder;
+import org.venice.beachfront.bfapi.auth.OpenIDTokenProvider;
+import org.venice.beachfront.bfapi.model.UserProfile;
+import org.venice.beachfront.bfapi.model.exception.UserException;
+import org.venice.beachfront.bfapi.services.UserProfileService;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Configuration
+@Controller
+@EnableOAuth2Client
+public class OAuth2Config {
+
+    // property set by spring-cloud-sso-connector
+    @Value("${ssoServiceUrl:placeholder}")
+	private String ssoServiceUrl;
+
+	@Value("${security.oauth2.client.clientId:placeholder}")
+	private String clientId;
+
+	@Value("${DOMAIN}")
+	private String domain;
+
+	@Value("${cookie.expiry.seconds}")
+	private int cookieExpirySeconds;
+
+	@Value("${cookie.name}")
+	private String cookieName;
+
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	@Autowired
+	private OAuth2ClientContext oAuth2ClientContext;
+
+	@Autowired
+	private OAuth2ProtectedResourceDetails resourceDetails;
+
+	@Autowired
+	private UserProfileService userProfileService;
+
+    @Bean
+    public OAuth2RestTemplate oauth2RestTemplate() {
+    	final OAuth2RestTemplate oauth2RestTemplate = new OAuth2RestTemplate(resourceDetails, oAuth2ClientContext);
+        oauth2RestTemplate.setAccessTokenProvider(accessTokenProviderChain());
+
+        return oauth2RestTemplate;
+    }
+
+    @Bean
+    public AccessTokenProvider accessTokenProviderChain() {
+        return new AccessTokenProviderChain(Arrays.<AccessTokenProvider> asList(
+            new OpenIDTokenProvider(),
+            new AuthorizationCodeAccessTokenProvider(),
+            new ImplicitAccessTokenProvider(),
+            new ResourceOwnerPasswordAccessTokenProvider(),
+            new ClientCredentialsAccessTokenProvider()));
+    }
+
+    @RequestMapping("/oauth/start")
+    public String authCode(final Model model, final HttpServletRequest request, final HttpServletResponse response) throws Exception {
+
+    	if (ssoServiceUrl.equals("placeholder")) {
```

Comment:
Maybe we should store this string literal in a central place and access it via Config.NULL_SSO_URL or something like that?

Not sure if it's used anywhere else.